internal/controller/datadogagent/component/agent/default.go

```diff
@@ -606,16 +605,37 @@ func envVarsForCoreAgent(dda metav1.Object) []corev1.EnvVar {
 func envVarsForTraceAgent(dda metav1.Object) []corev1.EnvVar {
 	envs := []corev1.EnvVar{
 		{
-			Name:  common.DDAPMInstrumentationInstallId,
-			Value: utils.GetDatadogAgentResourceUID(dda),
+			Name: common.DDAPMInstrumentationInstallId,
+			ValueFrom: &corev1.EnvVarSource{
+				ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "datadog-apm-telemetry-kpi",
```

Comment:
Since the cm name and keys are used multiple times in different places, can we put them in [constants](https://github.com/DataDog/datadog-operator/blob/main/pkg/constants/const.go) and reference from there?